.github/workflows/build.yml

```diff
@@ -0,0 +1,66 @@
+# Runs build related jobs.
+
+name: build
+
+on:
+  push:
+    branches: [main, next]
+  pull_request:
+    types: [opened, reopened, synchronize]
+
+jobs:
+  metal:
+    name: Build for metal
+    runs-on: ubuntu-latest
+    strategy:
+      fail-fast: false
+      matrix:
+        toolchain: [stable, nightly]
+    steps:
+      - uses: actions/checkout@main
+      - name: Build for metal
+        run: |
+          rustup update --no-self-update ${{ matrix.toolchain }}
+          make exec-metal
+
+  sve:
+    name: Build for sve
+    runs-on: ubuntu-latest
+    strategy:
+      fail-fast: false
+      matrix:
+        toolchain: [stable, nightly]
+    steps:
+      - uses: actions/checkout@main
+      - name: Build for sve
+        run: |
+          rustup update --no-self-update ${{ matrix.toolchain }}
+          make exec-sve
+  avx2:
+    name: Build for avx2
+    runs-on: ubuntu-latest
+    strategy:
+      fail-fast: false
+      matrix:
+        toolchain: [stable, nightly]
+    steps:
+      - uses: actions/checkout@main
+      - name: Build for avx2
+        run: |
+          rustup update --no-self-update ${{ matrix.toolchain }}
+          make exec-avx2
```

Comment:
I'm actually not sure if these are needed as the CI machine is unlikely to have the required hardware to build these properly (so, they will most likely just default to a standard build).